### PR TITLE
TT-7122 General Project to Burrito

### PR DESCRIPTION
**Pull request by gtryus**
- Implemented akuoBookToUsfm function to map Akuo-style book slots to USFM codes.
- Created projectDefaultToBurritoBookKey function for mapping project defaults to burrito book keys.
- Added comprehensive tests for both functions to ensure correct mapping and handling of edge cases.
- Updated BurritoBooks and useBurritoApmData to utilize the new mapping functions for improved book code resolution.

These changes enhance the handling of book codes in the application, ensuring accurate mappings and better integration with project defaults.

### CODE REVIEW

File: src/renderer/src/burrito/akuoBookToUsfm.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Maps a project's `book` default (Akuo A##/B## or a general-resource 3-digit code)
+ * to the burrito book key used in `burritoBooks`, export folder names, and section grouping.
+ */
+export function projectDefaultToBurritoBookKey(
+  projectBookDefault: string | undefined,
+  num2BookCode: (bookNum: number) => string | undefined
+): string | undefined {
+  const def = (projectBookDefault ?? 'B01').trim();
+  if (!def) return undefined;
```
Comment (sarahentzel) [inline src/renderer/src/burrito/akuoBookToUsfm.ts:29]: This can't ever by undefined.

Comment (sarahentzel) [reply src/renderer/src/burrito/akuoBookToUsfm.ts:29]: And actually, projectBookDefault isn't ever undefined because you're always passing in 'B01' everywhere.  That probably isn't necessary.

Comment (gtryus) [reply src/renderer/src/burrito/akuoBookToUsfm.ts:29]: ok, well I guess it won't hurt anything. I wonder if those lines were added because during loading, sometimes it comes back as undefined even though eventually, it won't.
